lib/msgpack/symbol.rb

```diff
@@ -14,6 +14,13 @@ def self.from_msgpack_ext(data)
     # The canonical way to do it for symbols would be:
     #  data.unpack1('A*').to_sym
     # However in this instance we can take a shortcut
-    data.to_sym
+
+    # We assume the string encoding is UTF-8, and let Ruby create either
+    # an ASCII symbol or UTF-8 symbol.
+    data.force_encoding(Encoding::UTF_8).to_sym
+  rescue EncodingError
+    # If somehow the string wasn't valid UTF-8 not valid ASCII, we fallback
+    # to what has been the historical behavior of creating a binary symbol
+    data.force_encoding(Encoding::BINARY).to_sym
```

Comment:
Here's the solution. I first try what should be the happy path, and deal with the error in a `rescue` as this should basically never happen ever, so no reason the happy path should pay an extra cost for it.

Reply:
Thanks for the fix and awesome code comment!